RestSharpClient/ApiClient/ApiService.cs

```diff
@@ -0,0 +1,242 @@
+            Response = Execute(restClient, request, apiMethodParams.Payload, apiMethodParams.Headers!);
+            return this;
+        }
+
+        public IApiService DeleteMethod(string uri, Dictionary<string, string>? headers = null)
+        {
+            var request = new RestRequest() { Resource = uri, Method = Method.Delete, RequestFormat = DataFormat.Json };
+            Response = Execute(restClient, request, null, headers!);
+            return this;
+        }
+
+        public IApiService DeleteMethod(ApiMethodParams apiMethodParams)
+        {
+            var request = new RestRequest() { Resource = apiMethodParams.Uri!, Method = Method.Delete, RequestFormat = DataFormat.Json };
+            request = SetRequestElements(request, apiMethodParams);
+            Response = Execute(restClient, request, apiMethodParams.Payload, apiMethodParams.Headers!);
+            return this;
+        }
+
+        private static RestResponse Execute(RestClient client, RestRequest request, object? requestObject = null, Dictionary<string, string>? headers = null)
+        {
+            RestResponse? response = null;
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                    request.AddHeader(header.Key, header.Value);
+            }
+
+            if (requestObject != null)
+                request.AddJsonBody(requestObject);
+
+
+            try {
+                response = client.Execute(request);
+            } catch (Exception exception) {
+                new ErrorLogger().LogError(exception, "");
+            }
+
+            return response!;
+        }
+
+        private static T Execute<T>(RestClient client, RestRequest request, object? requestObject = null, Dictionary<string, string>? headers = null)
+        {
+            RestResponse? response = null;
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                    request.AddHeader(header.Key, header.Value);
+            }
+
+            if (requestObject != null)
+                request.AddJsonBody(requestObject);
+
+
+            try
+            {
+                response = client.Execute(request);
+            }
+            catch (Exception exception)
+            {
+                new ErrorLogger().LogError(exception, "");
+            }
+
+            return new ApiService(ConfigurationHelper.BuildConfiguration().RestfulBooker.BaseUrl).GetValue<T>();
+        }
+
+
+        public T GetValue<T>()
+        {
+            return JsonConvert.DeserializeObject<T>(Response!.Content!, Converter.Settings)!;
+        }
+
+        public List<T> GetValueInList<T>()
+        {
+            return JsonConvert.DeserializeObject<List<T>>(Response!.Content!)!;
+        }
+
+        public RestResponse GetResponse()
+        {
+            return Response!;
+        }
+
+        public void Dispose()
+        {
+            throw new NotImplementedException();
```

Comment:
????

Reply:
Removed inheriting to IDisposable as mentioned on another comment and removed this.